lib/hello/shopping_cart.ex

```diff
@@ -0,0 +1,285 @@
+defmodule Hello.ShoppingCart do
+  @moduledoc """
+  The ShoppingCart context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Hello.Repo
+
+  #alias Hello.ShoppingCart.Cart
+  #alias Hello.ShoppingCart.CartItem
+
+  alias Hello.Catalog
+  alias Hello.ShoppingCart.{Cart, CartItem}
+
+  #Reloading the pruned cart to the caller
+  def prune_cart_items(%Cart{} = cart) do
+    {_, _} = Repo.delete_all(from(i in CartItem, where: i.cart_id == ^cart.id))
+    {:ok, reload_cart(cart)}
+  end
+
+  @doc """
+  Returns the list of carts.
+
+  ## Examples
+
+      iex> list_carts()
+      [%Cart{}, ...]
+
+  """
+  def list_carts do
+    Repo.all(Cart)
+  end
+
+  @doc """
+  Gets a single cart.
+
+  Raises `Ecto.NoResultsError` if the Cart does not exist.
+
+  ## Examples
+
+      iex> get_cart!(123)
+      %Cart{}
+
+      iex> get_cart!(456)
+      ** (Ecto.NoResultsError)
+
+  """
+  def get_cart!(id), do: Repo.get!(Cart, id)
+
+  @doc """
+  Creates a cart.
+
+  ## Examples
+
+      iex> create_cart(%{field: value})
+      {:ok, %Cart{}}
+
+      iex> create_cart(%{field: bad_value})
+      {:error, %Ecto.Changeset{}}
+
+  """
+
+  def get_cart_by_user_uuid(user_uuid) do
+    Repo.one(
+      from(c in Cart,
+        where: c.user_uuid == ^user_uuid,
+        left_join: i in assoc(c, :items),
```

Comment:
@keliumJU what is an association?

Reply:
returns a query with all the data associated with a structure, in this case a shopping cart has many items, and these items belong to another "CartItem" structure.